services/venice-server/src/main/java/com/linkedin/venice/listener/ListenerService.java

```diff
@@ -81,6 +84,16 @@ public ListenerService(
         serverConfig.getComputeQueueCapacity());
     new ThreadPoolStats(metricsRepository, computeExecutor, "storage_compute_thread_pool");
 
+    Optional<Executor> sslHandshakeExecutor = Optional.empty();
+    if (sslFactory.isPresent()) {
+      this.sslHandshakeExecutor = createThreadPool(
+          serverConfig.getSslHandshakeThreadPoolSize(),
+          "SSLHandShakeThread",
+          serverConfig.getSslHandshakeQueueCapacity());
```

Comment:
I think that's a good call since the code is in the server and version deployment to servers is a lot more work than config deployment if we encounter any unwanted/unexpected behavior.